netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -383,6 +385,20 @@ private void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
         return;
       }
 
+      if (headers.authority() == null) {
+        List<CharSequence> hosts = headers.getAll(HOST);
+        if (hosts.size() > 1) {
+          // RFC 7230 section 5.4
+          respondWithHttpError(ctx, streamId, 400, Status.Code.INTERNAL,
+              "Multiple host headers");
+          return;
+        }
+        if (!hosts.isEmpty()) {
+          headers.add(AUTHORITY.value(), hosts.get(0));
```

Comment:
Why not headers.authority(hosts.get(0))?

Reply:
Because it isn't implemented in the custom header implementation in GrpcHttp2HeadersUtils and I didn't feel like adding such a specialized call implementation.